### PR TITLE
multi plugin energy histogram with filter

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- refactor energy histogram that the usage as multi plugin is possible
- add `--<species>_energyHistogram.filter` (allow to filter particles)
- update documentaion

### CODE REVIEW

File: docs/TBG_macros.cfg [hunk 1]
```diff
@@ -134,8 +134,9 @@ TBG_<species>_pos_dbg="--<species>_position.period 1"
 
 
 # Create a particle-energy histogram [in keV] per species for every .period steps
-TBG_<species>_histogram="--<species>_energyHistogram.period 500 --<species>_energyHistogram.binCount 1024    \
```
Comment (ax3l) [inline docs/TBG_macros.cfg]: I think you will now need to update the following example's `.cfg` files:

- [ ] FoilLCT
- [ ] WarmCopper
- [ ] Bremsstrahlung
- [ ] WeibelTransverse
- [ ] Bunch
- [ ] Empty
- [ ] KelvinHelmholtz
- [ ] LaserWakefield

(from `grep -iR "energyHistogram" share/picongpu/examples/`)

File: docs/source/usage/plugins/energyHistogram.rst [hunk 2]
```diff
@@ -15,28 +15,48 @@ There are several command line parameters that can be used to set up this plugin
...
-=========================================== ======================================================================
+=========================================== =====================================================================================
 PIConGPU command line option                description
-=========================================== ======================================================================
+=========================================== =====================================================================================
 ``--e_energyHistogram.period``              The ouput periodicity of the **electron** histogram.
                                             A value of ``100`` would mean aoutput at simulation time step *0, 100, 200, ...*.
                                             If set to a non-zero value, the energy histogram of all **electrons** is computed.
                                             By default, the value is ``0`` and no histogram for the electrons is computed.
+``--e_energy.filter``                       Use filtered particles. All available filters will be shown with ``picongpu --help``
```
Comment (ax3l) [inline docs/source/usage/plugins/energyHistogram.rst]: hm, the user never interacts in our workflows with the `picongpu` binary and a reference to `picongpu --help` could be done on *any* of these files. I recommend to instead reference `particleFilters.param` here:

```
Available filters are set up in :ref:`particleFilters.param <usage-params-core>`.
```

File: docs/source/usage/plugins/energyHistogram.rst [hunk 3]
```diff
@@ -15,28 +15,48 @@ There are several command line parameters that can be used to set up this plugin
...
                                             Default is ``1024``.
 ``--e_energyHistogram.minEnergy``           Set the minimum energy for the **electron** histogram in *keV*.
                                             Default is ``0``, meaning *0 keV*.
 ``--e_energyHistogram.maxEnergy``           Set the maximum energy for the **electron** histogram in *keV*.
                                             There is **no default value**.
                                             This has to be set by the user if ``--e_energyHistogram.period 1`` is set.
-``--e_energyHistogram.distanceToDetector``  Distance in *meter* of a **electron** detector located far away in y direction with slit opening in x and z direction.
-                                            If set to *non-zero* value, only particles that would reach the detector are considered in the histogram.
-                                            Default ``0``, meaning all particles are considered in the **electron** histogram and no detector is assumed.
+``--e_energyHistogram.distanceToDetector``  Distance in *meter* of a **electron** detector located far away in y direction with
```
Comment (ax3l) [inline docs/source/usage/plugins/energyHistogram.rst]: oh fantastic, we can in a follow-up remove that horrible hackish option with a *filter* :)

File: docs/source/usage/plugins/energyHistogram.rst [hunk 4]
```diff
@@ -15,28 +15,48 @@ There are several command line parameters that can be used to set up this plugin
...
 ``--e_energyHistogram.slitDetectorZ``       Hight of the **electron** detector in *meter*.
                                             If not set, all particles are counted.
-=========================================== ======================================================================
+=========================================== =====================================================================================
+
+.. note::
+
+   This plugin is a multi plugin. 
+   Command line parameter can be used multiple times to create e.g. dumps with different dumping period.
+   In the case where a optional parameter with a default value is explicitly defined the parameter will be always passed to the instance of the multi plugin where the parameter is not set.
```
Comment (ax3l) [inline docs/source/usage/plugins/energyHistogram.rst]: a**n**

File: docs/TBG_macros.cfg (hunk 1 above)
Comment (psychocoderHPC) [reply docs/TBG_macros.cfg]: updated in the last commit

**Pull request merged**